### PR TITLE
:bug: Don't autosave gdocs on interval

### LINKED ISSUE #2196: Google Documents aren't publishing consistently
I suspect something to do with the `useAutoSaveDraft` is creating a race condition where the document can immediately unpublish itself after pressing Publish.

To reproduce: try publishing a document several times and occasionally it will unpublish again.

### PR DESCRIPTION
**Pull request by ikesau**
This should (almost entirely) fix #2196 which was caused by everyone's local 5 second autosave overwriting any updates to the DB.

Remaining edge cases:
### Scenario 1
1. Author A publishes a document
2. Author B updates the slug immediately after, unpublishing the document again

### Scenario 2
1. Author A updates the slug
2. Author B's local preview still has the old slug (because we don't auto-update this field)
3. Author B publishes the document, the old slug gets PUT back

These both seem very unlikely. Scenario 2 could be fixed with a little more complexity in the `synchronise content every 5 seconds` callback, but I don't think it's necessary to do.

### CODE REVIEW

**Comment by mlbrgl** [comment]
- [ ] copilot:summary
- [ ] is the branch up-to-date with master?
- [ ] what problem is being solved here?
- [ ] what solution has been chosen?
- [ ] are there other solutions?
- [ ] is the scope clear, well-defined and small?
- [ ] could a flow chart / sequence diagram / architecture diagram help?
- [ ] what is the happy path?
- [ ] could the code be easier to read and understand?
- [ ] can I help increase our confidence in the implementation by contributing tests?
- [ ] should some errors be made more prominent?
- [ ] what is missing, what should have been added / done / changed but hasn't?

File: adminSiteClient/GdocsPreviewPage.tsx [hunk 1]
```diff
@@ -52,9 +48,19 @@ export const GdocsPreviewPage = ({ match, history }: GdocsMatchProps) => {
...
+    const putCurrentGdoc = useDebounceCallback((current: OwidGdocInterface) => {
         cancelAllRequests()
-        setGdoc({ original: gdoc.original, current })
+        store.update(current)
+    }, 2000)
+    const setCurrentGdoc = (
+        current: OwidGdocInterface | undefined,
+        skipUpdatingDb = false
+    ) => {
+        setGdoc({ original: originalGdoc, current })
```
Comment (mlbrgl) [inline adminSiteClient/GdocsPreviewPage.tsx:59]: I think this messes up with the "Saving..." status. Is `originalGdoc` ever updated post page load?

**Review by mlbrgl — COMMENTED** [review]
I'm getting some instability, some of which might have been there before:
- there is a strange loop going on in the debug panel
- "saving..." stuck sometimes 

Overall, I think this auto-save / auto-refresh features might create more problems than potential value. See "alternative solutions" comment above.

**Comment by ikesau** [comment]
I think autorefresh is still quite useful, personally.

But I think I misinterpreted the existing saving logic, and maybe we should change that to make it less ambiguous.

i.e. Changing "Publish" to a "Save" button, and adding a new input labelled "Published" which is just a checkbox in the settings drawer. That way it's obvious when changes you make will get persisted to the DB, instead of things being contingent on whether the document is published.

Does that sound good?

**Comment by ikesau** [comment]
Yeah, exactly. I think autosaving is useful for like, word documents. But when we have 3 fields (`slug`, `publication context` and `publication date`) I think it's adding a lot of footgun ammunition for very little gain.

**Comment by mlbrgl** [comment]
Yes that makes sense.

On the additional checkbox, I think it would make for a more straightforward flow to change the label of the button, and perform a different committing action depending on the current state.

At the moment, the button has `Publish` and `Republish`, and we would have an additional `Save draft`. So the common actions remain accessible with one click, while the less common ones (unpublish, delete) are behind two.

What do you think?

**Comment by ikesau** [comment]
Okay I've tried to split the difference.

No more autosave, but still with a publish/republish button.

**Review by mlbrgl — APPROVED** [review]

**Pull request merged**